Calibration/IsolatedParticles/plugins/HcalHBHEMuonSimAnalyzer.cc

```diff
@@ -0,0 +1,521 @@
+#include <memory>
+#include <iostream>
+#include <vector>
+
+#include <TTree.h>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/EDAnalyzer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "FWCore/Common/interface/TriggerNames.h"
+#include "CommonTools/UtilAlgos/interface/TFileService.h"
+
+#include "DataFormats/EcalDetId/interface/EBDetId.h"
+#include "DataFormats/EcalDetId/interface/EEDetId.h"
+#include "DataFormats/HcalDetId/interface/HcalDetId.h"
+#include "SimDataFormats/CaloHit/interface/PCaloHitContainer.h"
+#include "SimDataFormats/Track/interface/SimTrack.h"
+#include "SimDataFormats/Track/interface/SimTrackContainer.h"
+#include "SimDataFormats/Vertex/interface/SimVertex.h"
+#include "SimDataFormats/Vertex/interface/SimVertexContainer.h"
+
+#include "Calibration/IsolatedParticles/interface/CaloPropagateTrack.h"
+#include "Calibration/IsolatedParticles/interface/eECALMatrix.h" 
+#include "Calibration/IsolatedParticles/interface/eHCALMatrix.h" 
+
+#include "Geometry/CaloGeometry/interface/CaloGeometry.h"
+#include "Geometry/Records/interface/CaloGeometryRecord.h"
+#include "Geometry/CaloGeometry/interface/CaloSubdetectorGeometry.h"
+#include "Geometry/CaloEventSetup/interface/CaloTopologyRecord.h"
+#include "Geometry/CaloTopology/interface/CaloSubdetectorTopology.h"
+#include "Geometry/CaloTopology/interface/HcalTopology.h"
+#include "Geometry/CaloTopology/interface/CaloTopology.h"
+#include "Geometry/HcalCommonData/interface/HcalHitRelabeller.h"
+#include "MagneticField/Engine/interface/MagneticField.h"
+#include "MagneticField/Records/interface/IdealMagneticFieldRecord.h"
+
+//#define EDM_ML_DEBUG
+
+class HcalHBHEMuonSimAnalyzer : public edm::one::EDAnalyzer<edm::one::WatchRuns,edm::one::SharedResources> {
+
+public:
+  explicit HcalHBHEMuonSimAnalyzer(const edm::ParameterSet&);
+  ~HcalHBHEMuonSimAnalyzer();
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+   
+private:
+  virtual void beginJob() override;
+  virtual void analyze(edm::Event const&, edm::EventSetup const&) override;
+  virtual void beginRun(edm::Run const&, edm::EventSetup const&) override;
+  virtual void endRun(edm::Run const&, edm::EventSetup const&) override {}
+  virtual void beginLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&) {}
+  virtual void endLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&) {}
```

Comment:
Please use `override` in all methods that you override. Otherwise you cause clang warnings due to inconsistent overrides.